src/broadcast/scrobblingmanager.cpp

```diff
@@ -0,0 +1,210 @@
+#include <QObject>
+
+#include "broadcast/scrobblingmanager.h"
+#include "broadcast/filelistener.h"
+#include "control/controlproxy.h"
+#include "engine/enginexfader.h"
+#include "mixer/deck.h"
+#include "mixer/playerinfo.h"
+#include "mixer/playermanager.h"
+
+ScrobblingManager::ScrobblingManager(PlayerManager *manager) :
+m_CPGuiTick("[Master]", "guiTick50ms",this),
+m_CPCrossfader("[Master]","crossfader", this),
+m_CPXFaderCurve(ConfigKey(EngineXfader::kXfaderConfigKey, 
+                "xFaderCurve"),this),
+
+m_CPXFaderCalibration(ConfigKey(EngineXfader::kXfaderConfigKey, 
+                      "xFaderCalibration"),this),
+
+m_CPXFaderMode(ConfigKey(EngineXfader::kXfaderConfigKey, 
+               "xFaderMode"),this),
+
+m_CPXFaderReverse(ConfigKey(EngineXfader::kXfaderConfigKey, 
+                  "xFaderReverse"),this),
+
+m_pManager(manager),
+m_broadcaster(new TrackTimers::ElapsedTimerQt)
+{
+    m_CPGuiTick.connectValueChanged(SLOT(slotGuiTick(double)));
+    connect(&PlayerInfo::instance(),SIGNAL(currentPlayingTrackChanged(TrackPointer)),
+            &m_broadcaster,SLOT(slotNowListening(TrackPointer)));
+    m_broadcaster
+        .addNewScrobblingService(new FileListener("nowListening.txt"));
+    startTimer(1000);
+}
+
+
+void ScrobblingManager::slotTrackPaused(TrackPointer pPausedTrack) {         
+    QMutexLocker locker(&m_mutex);
+    bool allPaused = true;
+    TrackInfo *pausedTrackInfo = nullptr;
+    for (TrackInfo *trackInfo : m_trackList) {
+        if (trackInfo->m_pTrack == pPausedTrack) {
+            pausedTrackInfo = trackInfo;
+            for (QString playerGroup : trackInfo->m_players) {
+                BaseTrackPlayer *player = m_pManager->getPlayer(playerGroup);
+                if (!player->isTrackPaused())
```

Comment:
check the pointer before using. If it is always valid ensure it by a VERIFY_OR_DEBUG_ASSERT.

Reply:
Alright.